server/src/main/java/io/crate/role/Role.java

```diff
@@ -76,13 +95,35 @@ public static Properties of(boolean login, GenericProperties<Object> properties)
                 }
             }
             JwtProperties jwtProperties = JwtProperties.fromMap(properties.getUnsafe(JWT_KEY));
-            return new Properties(login, hash, jwtProperties);
+
+            Map<String, Object> sessionSettings = new HashMap<>();
+            for (var p : properties.stream()
+                .filter(p -> !PASSWORD_KEY.equals(p.getKey()) && !JWT_KEY.equals(p.getKey())).toList()) {
+                SessionSetting<?> sessionSetting = sessionSettingRegistry.settings().get(p.getKey());
```

Comment:
We can remove this check, since we call explicitly the `validateSessionSettings` (and therefore the `SessionSettingsRegistry` from the signature.

On the otherhand, we could inline the validation here, and skip this extra method:
+: we avoid 2 iterations, one for validation, one for putting in the map
-: we add yet another argument (userName) the `of()` method of `Properties` which is weird

Maybe you have another suggestion to call `apply` and test the setters as well without the use of CoordinatorSessionSettings.

Reply:
Couldn't we have something like this here:

```java
            Map<String, Object> sessionSettings = new HashMap<>();
            for (var p : properties) {
                String property = p.getKey();
                if (property.equals(PASSWORD_KEY) || property.equals(JWT_KEY)) {
                    continue;
                }
                Object value = p.getValue();
                if (isReset == false) {
                    SessionSetting<?> sessionSetting = sessionSettingRegistry.settings().get(property);
                    assert sessionSetting != null : "sessionSetting shouldn't be null";
                    sessionSetting.validate(value);
                }
                sessionSettings.put(property, value);
            }
```

Note that:

- new `validate` method on `SessionSetting` (in exchange for not needing a dummy CoordinatorContext)
- no `stream().filter().toList()`, no need for allocations just to skip password/jwt properties
- `settings().get()` only when needed(?)